Remove Legacy Dashboard Support

LumioseSil opened the pull request:
#### e4325475d7d8b7fb7180384c6c288bfaac432b42
<pre>
Remove Legacy Dashboard Support
<a href="https://bugs.webkit.org/show_bug.cgi?id=255204">https://bugs.webkit.org/show_bug.cgi?id=255204</a>

Reviewed by Alexey Proskuryakov.

The Legacy Dashboard itself has not been used for many years.
We can remove it without issue now, especially since nothing
is dependent on the files themselves anymore.

* Source/WebCore/platform/mac/PasteboardMac.mm:
  (WebCore::Pasteboard::setDragImage):
* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/MigratedHeaders-input.xcfilelist:
* Source/WebKit/MigratedHeaders-output.xcfilelist:
* Source/WebKit/Modules/OSX_Private.modulemap:
* Source/WebKitLegacy/PlatformMac.cmake:
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/mac/Configurations/WebKitLegacy.xcconfig:
* Source/WebKitLegacy/mac/WebKit.order:
* Source/WebKitLegacy/mac/WebView/WebDashboardRegion.h: Removed.
* Source/WebKitLegacy/mac/WebView/WebDashboardRegion.mm: Removed.
* Source/WebKitLegacy/mac/WebView/WebUIDelegatePrivate.h:
* Source/WebKitLegacy/mac/WebView/WebView.mm:
  (-[WebView _addScrollerDashboardRegions:]): Deleted.
  (-[WebView _dashboardRegions]): Deleted.
  (-[WebView _setDashboardBehavior:to:]): Deleted.
  (-[WebView _dashboardBehavior:]): Deleted.
* Source/WebKitLegacy/mac/WebView/WebViewData.h:
* Source/WebKitLegacy/mac/WebView/WebViewPrivate.h:

Canonical link: <a href="https://commits.webkit.org/262798@main">https://commits.webkit.org/262798@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9868bb996b109f9154e78c884c224c4fd847b346

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/2625 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/2629 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/2762 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/4031 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/3016 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/2589 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/2768 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/2720 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/2314 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/2650 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/3111 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/2351 "Passed tests") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/3796 "Failed to compile WebKit") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/604 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/2335 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/3796 "Failed to compile WebKit") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/2708 "Built successfully") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/2355 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/3796 "Failed to compile WebKit") | 
| | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/2387 "Passed tests") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/2173 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/2391 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/2338 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/2377 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/310 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/2526 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov reviewed (approved):
Gtk failures look unrelated.

LumioseSil committed:
Remove Legacy Dashboard Support
https://bugs.webkit.org/show_bug.cgi?id=255204

Reviewed by Alexey Proskuryakov.

The Legacy Dashboard itself has not been used for many years.
We can remove it without issue now, especially since nothing
is dependent on the files themselves anymore.

* Source/WebCore/platform/mac/PasteboardMac.mm:
  (WebCore::Pasteboard::setDragImage):
* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/MigratedHeaders-input.xcfilelist:
* Source/WebKit/MigratedHeaders-output.xcfilelist:
* Source/WebKit/Modules/OSX_Private.modulemap:
* Source/WebKitLegacy/PlatformMac.cmake:
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/mac/Configurations/WebKitLegacy.xcconfig:
* Source/WebKitLegacy/mac/WebKit.order:
* Source/WebKitLegacy/mac/WebView/WebDashboardRegion.h: Removed.
* Source/WebKitLegacy/mac/WebView/WebDashboardRegion.mm: Removed.
* Source/WebKitLegacy/mac/WebView/WebUIDelegatePrivate.h:
* Source/WebKitLegacy/mac/WebView/WebView.mm:
  (-[WebView _addScrollerDashboardRegions:]): Deleted.
  (-[WebView _dashboardRegions]): Deleted.
  (-[WebView _setDashboardBehavior:to:]): Deleted.
  (-[WebView _dashboardBehavior:]): Deleted.
* Source/WebKitLegacy/mac/WebView/WebViewData.h:
* Source/WebKitLegacy/mac/WebView/WebViewPrivate.h:

Canonical link: https://commits.webkit.org/262798@main
